app/models/katello/repository.rb

```diff
@@ -288,6 +288,27 @@ def custom?
       !redhat?
     end
 
+    def empty_module_streams
+      repository_rpm = Katello::RepositoryRpm.table_name
+      repository_module_streams = Katello::RepositoryModuleStream.table_name
+      rpm = Katello::Rpm.table_name
+      module_stream = Katello::ModuleStream.table_name
+      module_stream_artifacts = Katello::ModuleStreamArtifact.table_name
+      module_stream_with_artifacts = ModuleStream.joins(
+        "INNER JOIN #{module_stream_artifacts} on #{module_stream_artifacts}.module_stream_id = #{module_stream}.id",
+        "INNER JOIN #{repository_module_streams} on #{repository_module_streams}.module_stream_id = #{module_stream}.id",
+        "INNER JOIN #{rpm} on #{rpm}.filename = #{module_stream_artifacts}.name",
+        "INNER JOIN #{repository_rpm} on #{repository_rpm}.rpm_id = #{rpm}.id").
+        where("#{repository_rpm}.repository_id" => self.id).
+        where("#{repository_module_streams}.repository_id" => self.id)
```

Comment:
Can we make any custom methods or scopes in `ModuleStream` or `ModuleStreamArtifact` to leverage the associations we have already? Do we need the custom query to associate the actual rpms to a module stream artifact?